tests/contrib/psycopg/test_psycopg_async.py

```diff
@@ -337,3 +338,26 @@ async def test_cursor_from_connection_shortcut(self):
         rows = await cur.fetchall()
         assert len(rows) == 1, rows
         assert rows[0][0] == "one"
+
+    async def test_cursor_async_connect_execute(self):
+        """Checks whether connection can execute operations with async iteration."""
+        async with psycopg.AsyncConnection.connect(**POSTGRES_CONFIG) as conn:
+            async with conn.cursor() as cur:
+                await cur.execute("DROP TABLE IF EXISTS {}".format(TEST_TABLE))
+                await cur.execute(
+                    """CREATE TABLE {} (
+                            a INT,
+                            b VARCHAR(32)
+                            )
+                            """.format(
+                        TEST_TABLE
+                    )
+                )
+                await cur.execute("INSERT INTO {} (a, b) VALUES (1, 'aa');".format(TEST_TABLE))
+                await cur.execute("INSERT INTO {} (a, b) VALUES (2, 'bb');".format(TEST_TABLE))
+                await cur.execute("INSERT INTO {} (a, b) VALUES (3, 'cc');".format(TEST_TABLE))
+
+                count = 1
+                async for row in cur:
+                    assert row[0] == count
+                    count += 1
```

Comment:
Can we validate the spans generate by this test case?
Something like 
```suggestion
                    count += 1
                    spans = self.get_spans()
                    self.assertEqual(len(spans), X)
                    # maybe some assertions on span names or resource names
```